src/main/java/seedu/jimi/logic/parser/Frequency.java

```diff
@@ -0,0 +1,170 @@
+                freqWord = freqStrs[1];
+                numberOfTimes = StringToInt.parse(freqStrs[2]);
+            }
+            break;
+        default:
+            throw new IllegalValueException(MESSAGE_INVALID_FREQUENCY_FIELD);
+        }      
+    }
+    
+    public String getFreqWord() {
+        return freqWord;
+    }
+    
+    public int getFreqQuantifier() {
+        return freqQuantifier;
+    }
+    
+    public int getNumberOfTimes() {
+        return numberOfTimes;
+    }
+    
+    public static boolean isMinute(final String str) {
+        return minuteEquivalents.contains(str);
+    }
+    
+    public static boolean isHour(final String str) {
+        return hourEquivalents.contains(str);
+    }
+    
+    public static boolean isDay(final String str) {
+        return dayEquivalents.contains(str);
+    }
+    
+    public static boolean isDayOfWeek(final String str) {
+        return dayOfWeekEquivalents.contains(str);
+    }
+    
+    public static String getDayOfWeekFull(final String str) throws IllegalValueException {
+        String lowerStr = str.trim().toLowerCase();
+        if(mondayEquilavents.contains(lowerStr))        return "MONDAY";
+        else if(tuesdayEquivalent.contains(lowerStr))   return "TUESDAY";
+        else if(wednesdayEquivalent.contains(lowerStr)) return "WEDNESDAY";
+        else if(thursdayEquivalent.contains(lowerStr))  return "THURSDAY";
+        else if(fridayEquivalent.contains(lowerStr))    return "FRIDAY";
+        else if(saturdayEquivalent.contains(lowerStr))  return "SATURDAY";
+        else if(sundayEquivalent.contains(lowerStr))    return "SUNDAY";
+        else if(weekendEquivalents.contains(lowerStr))  return "SATURDAY";
+        else throw new IllegalValueException("Wrong week of day input!");
+    }
+    
+    public static boolean isWeek(final String str) {
+        return weekEquivalents.contains(str.toLowerCase());
+    }
+    
+    public static boolean isMonth(final String str) {
+        return monthEquivalents.contains(str.toLowerCase());
+    }
+    
+    public static boolean isYear(final String str) {
+        return yearEquivalents.contains(str.toLowerCase());
+    }
+    
+    
+    public static Date getNextDate(final Date date, final int quantifier, final String freq) 
+            throws IllegalValueException {
+        LocalDateTime ldt = LocalDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault());
+        if(isMinute(freq))     ldt = ldt.plusMinutes(quantifier);
+        else if(isHour(freq))  ldt = ldt.plusHours(quantifier);
+        else if(isDay(freq))   ldt = ldt.plusDays(quantifier);
+        else if(isWeek(freq))  ldt = ldt.plusWeeks(quantifier);
+        else if(isMonth(freq)) ldt = ldt.plusMonths(quantifier);
+        else if(isYear(freq))  ldt = ldt.plusYears(quantifier);
+        else throw new IllegalValueException(MESSAGE_INVALID_FREQUENCY_NAME);
+        
+        return Date.from(ldt.atZone(ZoneId.systemDefault()).toInstant());
```

Comment:
i personally dont mind this codestyle, but i think they hate it. they want braces on everything.